### PR TITLE
Use bytes for Ed25519Signature public key

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Looks like using the `PublicKey` type here was a mistake, as the validation is not intended to be at the syntactic level. Unfortunately this is a breaking change, and there's no way to avoid that.

## Links to any relevant issues

https://github.com/iotaledger/inx-chronicle/issues/1308

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by Thoralf-M — CHANGES_REQUESTED** [review]
Missing changelog entry and version update or don't you need a new release now?

**Comment by DaughterOfMars** [comment]
Yeah, I figured I'd wait until we decide whether this is even happening 😆

File: sdk/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,20 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+## 1.1.4 - 2024-MM-DD
+
+### Added
+
+- `Ed25519Signature` methods `new_from_bytes`, `public_key_bytes`, `from_bytes`, `try_verify`;
+
+### Changed
```
Comment (thibault-martinez) [inline sdk/CHANGELOG.md]: There is a Deprecated section in the template so I'd use it

File: bindings/core/src/method_handler/utils.rs [hunk 2]
```diff
@@ -94,9 +94,10 @@ pub(crate) fn call_utils_method_internal(method: UtilsMethod) -> Result<Response
             Response::Ok
         }
         UtilsMethod::VerifyEd25519Signature { signature, message } => {
+            use iota_sdk::types::block::Error;
```
Comment (thibault-martinez) [inline bindings/core/src/method_handler/utils.rs]: Move to the other imports pls

File: sdk/src/types/block/signature/ed25519.rs [hunk 3]
```diff
@@ -31,24 +31,44 @@ impl Ed25519Signature {
...
+        }
     }
 
     /// Returns the public key of an [`Ed25519Signature`].
+    #[deprecated(since = "1.1.4", note = "use Ed25519Signature::public_key_bytes instead")]
     pub fn public_key(&self) -> &PublicKey {
+        panic!("deprecated method: use Ed25519Signature::public_key_bytes instead")
+    }
+
+    /// Returns the public key of an [`Ed25519Signature`].
```
Comment (Alex6323) [inline sdk/src/types/block/signature/ed25519.rs]:
```suggestion
    /// Returns the unvalidated public key bytes of an [`Ed25519Signature`].
```

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**